### PR TITLE
Capture more information on errors.

### PR DESCRIPTION
**Pull request by johndvorakHR**

### CODE REVIEW

**Comment by johndvorakHR** [comment]
Improved so it doesn't break when all tests pass :+1:

File: index.js [hunk 1]
```diff
@@ -11,35 +11,40 @@ function TaserReporter(baseReporterDecorator, config) {
...
-        };
-    }
-    
-    function onBrowserError(browser, error) {
-        errors[browser.id] = errors[browser.id] || [];
-        errors[browser.id].push(error);
+
+    function onBrowserStart(browser, error) {
+        if (typeof error == 'string') {
+            console.log('error', error)
```
Comment (Spidy88) [inline index.js]: Please remove console logs

File: index.js [hunk 2]
```diff
@@ -11,35 +11,40 @@ function TaserReporter(baseReporterDecorator, config) {
     // This is not a full list of events that can be listened for, only the set we currently care about
     this.onRunStart = onRunStart.bind(this);
     this.onBrowserStart = onBrowserStart.bind(this);
-    this.onBrowserError = onBrowserError.bind(this);
```
Comment (Spidy88) [inline index.js:14]: Why was onBrowserStart and onBrowserError merged? Now they have if conditions and it makes the code tougher to understand.

File: index.js [hunk 3]
```diff
@@ -11,35 +11,40 @@ function TaserReporter(baseReporterDecorator, config) {
...
-            },
-            errors: errors[browser.id] || []
-        };
-    }
-    
-    function onBrowserError(browser, error) {
-        errors[browser.id] = errors[browser.id] || [];
-        errors[browser.id].push(error);
+
+    function onBrowserStart(browser, error) {
```
Comment (Spidy88) [inline index.js]: If onBrowserStart ever hands in a second parameter that is a string and not actually an onBrowserError event, this code will be tough to debug. They should be separate functions

File: index.js [hunk 4]
```diff
@@ -54,6 +59,18 @@ function TaserReporter(baseReporterDecorator, config) {
             category = 'failed';
         }
 
+        if (testResult.log.length && testResult.log.length > 0) {
```
Comment (Spidy88) [inline index.js]: What is the point of the first `testResult.log.length` ?

**Comment by johndvorakHR** [comment]
Updated

File: index.js [hunk 5]
```diff
@@ -19,27 +19,39 @@ function TaserReporter(baseReporterDecorator, config) {
...
-        };
-    }
-    
+
+    function makeBrowserRecord (id, name, fullName) {
+        if (errors[id] == null) {
+            errors[id] = [];
+        }
+
+        if (browsers[id] == null) {
```
Comment (Spidy88) [inline index.js]: Triple equals please

File: index.js [hunk 6]
```diff
@@ -19,27 +19,39 @@ function TaserReporter(baseReporterDecorator, config) {
...
-                ignored: [],
-                failed: []
-            },
-            errors: errors[browser.id] || []
-        };
-    }
-    
+
+    function makeBrowserRecord (id, name, fullName) {
+        if (errors[id] == null) {
```
Comment (Spidy88) [inline index.js]: Triple equals please

File: index.js [hunk 7]
```diff
@@ -19,27 +19,39 @@ function TaserReporter(baseReporterDecorator, config) {
...
+                    ignored: [],
+                    failed: []
+                },
+                errors: errors[id]
+            };
+        }
+    } 
+
     function onBrowserError(browser, error) {
+        makeBrowserRecord(browser.id, browser.name, browser.fullName);
```
Comment (Spidy88) [inline index.js:45]: Way better. nice

**Comment by johndvorakHR** [comment]
Updated

**Pull request merged**